decisions/2021/02/03-fe-meeting.md

```diff
@@ -0,0 +1,97 @@
+**Date**: 2021-02-03
+
+**Attending**: Austin, Birk, Bjørn, Christos, Debora, Edoardo, Hendrik,
+Ismay, Jan Henrik, Ismay, Jan-Gerke, Jennifer, Joakim, Joe, Kai, Martin,
+Médi-Remi, Milagros
+
+**Next Meeting**: February 17, 2021 @ 15:00
+
+# Agenda
+
+## General updates
+
+
+## Working group updates
+
+-   **Cypress**
+    
+    Finishing up the PRs from December, one for codemods to help with
+    upgrading consuming code bases automatically, another for the
+    network shim functionality.
+
+-   **PWA & Offline/Responsive Dashboard**
+
+    Responsive: finished the implementation, awaiting testing to
+    finalize.
+
+    Some learnings surrounding on switching to a different component
+    causes re-renders.
+
+    Offline: changes coming in soon, so they can be tested.
+
+    Headerbar will need changes in terms of responsiveness _and_ in
+    terms of offline compability.
+
+-   **Plugin builds**
+
+    Based around dynamic modules, which is its own point.
+
+-   **UI & Platform updates**
+
+    Looking into the "drifting Tooltip" problem.
+
+    Might be a fix possible in the Popper modifiers.
+
+-   **Developer portal proposal and community plans**
+
+    Revamp the developer hub to have a single place for all the
+    developer docs, including blogs, guides, etc.
+
+    Docusaurus seems like a possible candidate to replace Docsite.
+
+    Needs a decision around Docusaurus.
+
+    Developer outreach by meetups, inviting external devs to Slack, etc.
+
+    Move forward with Docusaurus v2, try it out in cli-utils-docsite by
+    replacing what we currently have, and see how the "default" render
+    looks, then trying to roll it out to other libraries, see how that
+    goes.
```

Comment:
@deboragaleano @varl I know this is what we discussed, but I think it's more practical to try Docusaurus in the dhis2.github.io repo first, then incrementally migrate some of the reference documentation sites

Reply:
Different reading of the decision made in the meeting and thoughts since then, we can leave this in the notes if we want

Reply:
That works too. I was considering that converting Jekyll to Docusaurus is guaranteed to more of less "just work", so the outstanding question in terms of research is if Docusaurus works as a Docsify replacement in `cli-utils-docsite`.

Perhaps we aren't trying to find one tool that does both? I think that would be helpful in the long run, though I don't know what else has been discussed here when I was on leave.

Reply:
Migrating from Jekyll would be indeed an easy task. Ideally, it would be great if it can work as a replacement in `cli-utils-docsite` as well -- though I'd need to do more research on how to practically implement it.

Reply:
I think the challenge here is the "wrapper" tool of `cli-utils-docsite` which takes some designing and work to make "out of the box", so the more critical research pieces to me (in order) are:

1. Can we write the type of documentation we want in Docusaurus, even if that is a standalone thing at developers.dhis2.org (i.e. reference docs but also guides that span multiple tools)
2. Can we dynamically pull markdown / mdx files from remote places at build time (or runtime) so that we can colocate the doc source with the relevant repos?  And can we do this with independent versioning as well?
3. Can we package a stripped-down version of docusaurus in `cli-utils-docsite` to enable local previews of the rendered documentation (particularly reference documentation) within some repo?  I don't see this as super critical when we're talking about standard `.md` and `.mdx` files, but it is a nice-to-have

Reply:
I think the answer to "can we" is yes to all 3, they just require different amounts and types of work

Reply:
I'm OK with that list of priorities.